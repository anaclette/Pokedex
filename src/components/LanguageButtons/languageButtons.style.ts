import {StyleSheet} from 'react-native';
import colors from '../../themes/colors';
import metrics from '../../themes/metrics';

export const styles = StyleSheet.create({
  container: {
    position: 'absolute',
    top: '8%',
    right: 0,
    zIndex: 1,
  },
  languageOptionWrapper: {
    borderColor: colors.translucidLightBackground,
    borderRadius: metrics.scale(20),
    borderWidth: metrics.scale(1),
    marginVertical: metrics.verticalScale(1),
  },
  flagImage: {
    width: metrics.moderateScale(30, 0.4),
    height: metrics.moderateScale(30, 0.4),
    borderRadius: metrics.scale(20),
  },
});
