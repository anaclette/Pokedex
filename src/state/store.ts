import {combineReducers, configureStore} from '@reduxjs/toolkit';
import AsyncStorage from '@react-native-async-storage/async-storage';
import {persistReducer, persistStore} from 'redux-persist';
import authReducer from './reducers/authReducer';
import pokemonsReducer from './reducers/pokemonsReducer';
import favouritesReducer from './reducers/favouritesReducer';
import {STATE_MODULES} from '../common/constants';
import {i18nReducer} from './reducers/i18nReducer';

const persistConfig = {
  key: 'root',
  storage: AsyncStorage,
};

const rootReducer = combineReducers({
  [STATE_MODULES.AUTH]: authReducer,
  [STATE_MODULES.POKEMONS]: pokemonsReducer,
  [STATE_MODULES.FAVOURITES]: favouritesReducer,
  [STATE_MODULES.LANGUAGES]: i18nReducer.reducer,
});

const persistedReducer = persistReducer(persistConfig, rootReducer);

const store = configureStore({
  reducer: persistedReducer,
  middleware: getDefaultMiddleware =>
    getDefaultMiddleware({
      immutableCheck: false,
      serializableCheck: false,
    }),
});

const persistor = persistStore(store);

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;

export {persistor, store};
