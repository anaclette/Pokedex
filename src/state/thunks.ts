import {createAsyncThunk} from '@reduxjs/toolkit';
import {ACTIONS, baseUrl, STATE_MODULES} from '../common/constants';

export const getPokemons = createAsyncThunk(
  `${STATE_MODULES.POKEMONS}/${ACTIONS.GET_POKEMONS}`,
  async (_, {rejectWithValue}) => {
    try {
      const pokemons = await fetch(
        `${baseUrl}/${STATE_MODULES.POKEMONS}?limit=50&offset=50`,
      )
        .then(response => response.json())
        .then(data => data);
      return pokemons;
    } catch (error) {
      rejectWithValue(error);
    }
  },
);
