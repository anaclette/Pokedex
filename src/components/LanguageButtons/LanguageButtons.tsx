import React from 'react';
import {useTranslation} from 'react-i18next';
import {Image, View} from 'react-native';
import {useAppDispatch, useAppSelector} from '../../state/hooks';
import {switchLanguage} from '../../state/reducers/i18nReducer';
import colors from '../../themes/colors';
import Button from '../Button';
import {styles} from './languageButtons.style';
import {TranslationKeys} from '../../locale/translations/keys';

export const LanguageButtons = () => {
  const dispatch = useAppDispatch();
  const {i18n, t} = useTranslation();
  const languageOptions = useAppSelector(
    state => state.languages.supportedLangs,
  );
  return (
    <View style={styles.container}>
      <Button
        style={styles.languageOptionWrapper}
        accessibilityLabel={t(TranslationKeys.FLAG_BUTTON_IMAGE)}
        accessibilityRole="button"
        activeOpacity={0.5}
        underlayColor={colors.transparent}
        onPress={() => {
          dispatch(switchLanguage(languageOptions.es));
          i18n.changeLanguage(languageOptions.es);
        }}
        children={
          <Image
            style={styles.flagImage}
            source={require('../../assets/images/argentina_flag.jpeg')}
          />
        }
      />

      <Button
        style={styles.languageOptionWrapper}
        accessibilityLabel={t(TranslationKeys.FLAG_BUTTON_IMAGE)}
        accessibilityRole="button"
        activeOpacity={0.5}
        underlayColor={colors.transparent}
        onPress={() => {
          dispatch(switchLanguage(languageOptions.en));
          i18n.changeLanguage(languageOptions.en);
        }}
        children={
          <Image
            style={styles.flagImage}
            source={require('../../assets/images/USA_flag.webp')}
          />
        }
      />
    </View>
  );
};
