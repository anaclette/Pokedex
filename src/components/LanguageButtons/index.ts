import {LanguageButtons} from './LanguageButtons';

export default LanguageButtons;
