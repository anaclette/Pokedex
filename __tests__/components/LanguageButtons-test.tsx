import React from 'react';
import {Provider} from 'react-redux';
import {render} from '@testing-library/react-native';
import LanguageButtons from '../../src/components/LanguageButtons';
import {store} from '../../src/state/store';

const component = (
  <Provider store={store}>
    <LanguageButtons />
  </Provider>
);

test('language buttons render correctly', () => {
  const {getAllByRole, getAllByLabelText} = render(component);
  expect(component).toMatchSnapshot();

  const button = getAllByRole('button');
  expect(button).toBeTruthy();

  const label = getAllByLabelText('EEUU flag image to choose english language');
  expect(label).toBeTruthy();
});
