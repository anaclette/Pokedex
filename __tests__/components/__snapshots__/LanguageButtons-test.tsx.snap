// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`language buttons render correctly 1`] = `
<Provider
  store={
    Object {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <LanguageButtons />
</Provider>
`;
